src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -149,6 +149,17 @@ public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences
             }
         }
 
+        internal override void AddSynthesizedAttributes(ModuleCompilationState compilationState, ref ArrayBuilder<SynthesizedAttributeData> attributes)
+        {
+            // Make sure we emit [Dynamic] on synthesized parameter symbols when the original parameter was
+            // dynamic in order to facilitate debugging.
+            if (this.Type.ContainsDynamic())
```

Comment:
We making sure the attributes are available only for lambda parameters, this check is not that strict. Either the attributes should be optional, or the check should be tighter, in my opinion.